executor/distsql.go

```diff
@@ -626,27 +626,28 @@ func (e *IndexLookUpExecutor) startWorkers(ctx context.Context, initBatchSize in
 func (e *IndexLookUpExecutor) needPartitionHandle(tp getHandleType) (bool, error) {
 	var col *expression.Column
 	var needPartitionHandle, ret bool
+	needPartitionHandle = e.index.Global || (e.partitionTableMode && e.keepOrder)
 	if tp == getHandleFromIndex {
 		cols := e.idxPlans[0].Schema().Columns
 		outputOffsets := e.dagPB.OutputOffsets
 		col = cols[outputOffsets[len(outputOffsets)-1]]
-		needPartitionHandle = e.index.Global || (e.partitionTableMode && e.keepOrder)
 		ret = col.ID == model.ExtraPhysTblID || col.ID == model.ExtraPidColID
 	} else {
 		cols := e.tblPlans[0].Schema().Columns
 		outputOffsets := e.tableRequest.OutputOffsets
 		col = cols[outputOffsets[len(outputOffsets)-1]]
 
-		needPartitionHandle = e.index.Global || (e.partitionTableMode && e.keepOrder)
 		// no ExtraPidColID here, because TableScan shouldn't contain them.
 		ret = col.ID == model.ExtraPhysTblID
 	}
 
 	// TODO: fix global index related bugs later
-	if needPartitionHandle != ret && !e.index.Global {
-		return ret, errors.Errorf("Internal error, needPartitionHandle(%t) != ret(%t), tp(%d)", needPartitionHandle, ret, tp)
+	// For `SelectLock`, the `ExtraPhysTblID` will contained in schema,
+	// but it needn't partition handle to keep order.
+	if needPartitionHandle && !ret && !e.index.Global {
+		return ret, errors.Errorf("Internal error, needPartitionHandle != ret, tp(%d)", tp)
 	}
-	return ret, nil
+	return needPartitionHandle, nil
```

Comment:
`ret` is declared but not used?

Reply:
This is designed to check status, but this PR not perfect now. I still working on it.